docs/0.1/grid_product_catalog_family_specification.md

```diff
@@ -0,0 +1,496 @@
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_update_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogUpdateAction must include:
+
+- Address of the agent submitting the transaction
+- Address of the organization the catalog is being created for
+- Address of the catalog to be updated
+
+The outputs for CatalogUpdateAction must include:
+
+- Address of the catalog to be updated
+
+#### CatalogDeleteAction
+
+CatalogDeleteAction deletes an existing catalog from state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf
+of the organization that corresponds to the owner in the delete transaction.
+(Organizations and agents are defined by the Pike smart contract.)
+
+```protobuf
+message CatalogDeleteAction {
+    string owner = 1;
+    string catalog_id = 2;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is valid, otherwise it's
+  invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_delete_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogDeleteAction must include:
+
+- Address of the agent submitting the transaction
+- Address of the organization the catalog is being created for
+- Address of the catalog to be deleted
+
+The outputs for CatalogDeleteAction must include:
+
+- Address of the catalog to be deleted
+
+**_NOTE: Deleting a catalog is potentially dangerous operation that could leave
+dangling references and should be done with care._**
+
+### Catalog Product Actions
+
+#### CatalogProductCreateAction
+
+The CatalogProductCreateAction adds a new catalog_product to state. The
+catalog_product references a Grid Product for the shared item level master
+data. The transaction should be submitted by an agent, which is identified by
+its signing key, acting on behalf of the organization that corresponds to the
+owner in the create transaction. (Organizations and agents are defined by the
+Pike smart contract.)
+
+```protobuf
+message CatalogProductCreateAction {
+    string catalog_id = 1
+    string product_id = 2
+    repeated PropertyValues properties = 3;
+}
+```
+
+Validation requirements:
+
+- If a catalog product with product ID and catalog ID already exists the
+  transaction is invalid.
+- If the Grid product the catalog product is referencing does not exist in
+  state the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_product for the organization,
```

Comment:
For which org, the Cat owner or the Product owner...

Reply:
In Pike, Agents can only be members of a single organization, which means that all of the permissions they have apply only to that organization.